x-pack/solutions/security/plugins/elastic_assistant/server/routes/attack_discovery/schedules/update.test.ts

```diff
@@ -0,0 +1,126 @@
+import { updateAttackDiscoverySchedulesRequest } from '../../../__mocks__/request';
+import { getInternalAttackDiscoveryScheduleMock } from '../../../__mocks__/attack_discovery_schedules.mock';
+import { AttackDiscoveryScheduleDataClient } from '../../../lib/attack_discovery/schedules/data_client';
+
+const { clients, context } = requestContextMock.createTools();
+const server: ReturnType<typeof serverMock.create> = serverMock.create();
+clients.core.elasticsearch.client = elasticsearchServiceMock.createScopedClusterClient();
+
+const updateAttackDiscoverySchedule = jest.fn();
+const mockSchedulingDataClient = {
+  findSchedules: jest.fn(),
+  getSchedule: jest.fn(),
+  createSchedule: jest.fn(),
+  updateSchedule: updateAttackDiscoverySchedule,
+  deleteSchedule: jest.fn(),
+  enableSchedule: jest.fn(),
+  disableSchedule: jest.fn(),
+} as unknown as AttackDiscoveryScheduleDataClient;
+const mockApiConfig = {
+  connectorId: 'connector-id',
+  actionTypeId: '.bedrock',
+  model: 'model',
+  provider: OpenAiProviderType.OpenAi,
+};
+const mockRequestBody: UpdateAttackDiscoverySchedulesRequestBody = {
+  name: 'Test Schedule 2',
+  schedule: {
+    interval: '15m',
+  },
+  params: {
+    alertsIndexPattern: '.alerts-security.alerts-default',
+    apiConfig: mockApiConfig,
+    end: 'now',
+    size: 50,
+    start: 'now-24h',
+  },
+  actions: [],
+};
+
+describe('updateAttackDiscoverySchedulesRoute', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+    context.elasticAssistant.getAttackDiscoverySchedulingDataClient.mockResolvedValue(
+      mockSchedulingDataClient
+    );
+    context.elasticAssistant.getRegisteredFeatures.mockReturnValue({
+      ...defaultAssistantFeatures,
+      assistantAttackDiscoverySchedulingEnabled: true,
+    });
+    updateAttackDiscoverySchedulesRoute(server.router);
+    updateAttackDiscoverySchedule.mockResolvedValue(
+      getInternalAttackDiscoveryScheduleMock(mockRequestBody)
+    );
+  });
+
+  it('should handle successful request', async () => {
+    const response = await server.inject(
+      updateAttackDiscoverySchedulesRequest('schedule-1', mockRequestBody),
+      requestContextMock.convertContext(context)
+    );
+    expect(response.status).toEqual(200);
+    expect(response.body).toEqual(expect.objectContaining({ ...mockRequestBody }));
+  });
+
+  it('should handle missing data client', async () => {
+    context.elasticAssistant.getAttackDiscoverySchedulingDataClient.mockResolvedValue(null);
+    const response = await server.inject(
+      updateAttackDiscoverySchedulesRequest('schedule-2', mockRequestBody),
+      requestContextMock.convertContext(context)
+    );
+
+    expect(response.status).toEqual(500);
+    expect(response.body).toEqual({
+      message: 'Attack discovery data client not initialized',
+      status_code: 500,
+    });
+  });
+
+  it('should handle `dataClient.createSchedule` error', async () => {
```

Comment:
nit:  Consider `createSchedule` -> `updateSchedule`

Reply:
done in https://github.com/elastic/kibana/pull/216656/commits/89adba597b631e0c7d4a055927a7fb2543cc61b7